specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/ManagedLedgerDigestUploads.json

```diff
@@ -0,0 +1,335 @@
+          },
+          "202": {
+            "description": "Disabling the ledger digest upload is in progress."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Disables uploading ledger digests for a database": {
+            "$ref": "./examples/ManagedLedgerDigestUploadsDisable.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ManagedLedgerDigestUploads": {
+      "description": "Azure SQL Database ledger digest upload settings.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ManagedLedgerDigestUploadsProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ManagedLedgerDigestUploadsListResult": {
+      "description": "A list of ledger digest upload settings.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedLedgerDigestUploads"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ManagedLedgerDigestUploadsProperties": {
+      "description": "The properties of a database ledger digest upload settings.",
+      "type": "object",
+      "properties": {
+        "digestStorageEndpoint": {
+          "description": "The digest storage endpoint, which must be either an Azure blob storage endpoint or an URI for Azure Confidential Ledger.",
+          "type": "string"
+        },
+        "state": {
+          "description": "Specifies the state of ledger digest upload.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "LedgerDigestUploadsState",
+            "modelAsString": false
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "DatabaseNameParameter": {
+      "name": "databaseName",
+      "in": "path",
+      "description": "The name of the database.",
+      "required": true,
+      "type": "string",
```

Comment:
It is good to add RegEx based parameter's name validation, that you have/will have on server side. For example, whether it shouldn't be empty, have only alpha-numeric characters etc. See: [https://github.com/Azure/azure-openapi-validator/blob/main/docs/resource-name-restriction.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/resource-name-restriction.md).

Reply:
We have all sorts of error handlers in the server side. We also have validator in PreFlight